homeassistant/components/mazda/__init__.py

```diff
@@ -109,34 +109,12 @@ async def async_handle_service_call(service_call: ServiceCall) -> None:
         if vehicle_id == 0 or api_client is None:
             raise HomeAssistantError("Vehicle ID not found")
 
-        if service_call.service in (
-            "start_engine",
-            "stop_engine",
-            "turn_on_hazard_lights",
-            "turn_off_hazard_lights",
-        ):
-            _LOGGER.warning(
-                "The mazda.%s service is deprecated and has been replaced by a button entity; "
-                "Please use the button entity instead",
-                service_call.service,
-            )
-
-        if service_call.service in ("start_charging", "stop_charging"):
-            _LOGGER.warning(
-                "The mazda.%s service is deprecated and has been replaced by a switch entity; "
-                "Please use the charging switch entity instead",
-                service_call.service,
-            )
-
         api_method = getattr(api_client, service_call.service)
         try:
-            if service_call.service == "send_poi":
-                latitude = service_call.data["latitude"]
-                longitude = service_call.data["longitude"]
-                poi_name = service_call.data["poi_name"]
-                await api_method(vehicle_id, latitude, longitude, poi_name)
-            else:
-                await api_method(vehicle_id)
+            latitude = service_call.data["latitude"]
```

Comment:
Side note: We can move the service data access up out of the try... except block. We only want the line that can raise here.